src/core/stream/representation/get_needed_segments.ts

```diff
@@ -369,3 +391,71 @@ function isEndGarbageCollected(
 
   return false;
 }
+
+/**
+ * Returns `true` if a segment that has been garbage-coolected at the start
+ * might profit from being re-loaded.
+ *
+ * Returns `false` if we have a high chance of staying in the same situation
+ * after re-loading the segment.
+ *
+ * This function takes in argument the entries of a SegmentBuffer's history
+ * related to the corresponding asegment and check if the segment appeared
+ * garbage-collected at the start directly after the last few times it was
+ * pushed, indicating that the issue might be sourced at a browser issue instead
+ * of classical garbage collection.
+ *
+ * @param {Array.<Object>} segmentEntries
+ * @returns {boolean}
+ */
+function shouldReloadSegmentGCedAtTheStart(
+  segmentEntries : IBufferedHistoryEntry[]
+) : boolean {
+  let prevSegmentBufferStart : IBufferedHistoryEntry | null = null;
+  for (let entryIdx = segmentEntries.length - 1; entryIdx >= 0; entryIdx--) {
+    const entry = segmentEntries[entryIdx];
+    if (prevSegmentBufferStart !== null) {
+      const prevBufferedStart = prevSegmentBufferStart.bufferedStart;
+      if (Math.abs(prevBufferedStart - entry.bufferedStart) <= 0.01) {
```

Comment:
Just by curiosity by why did you choose that value (`0.01`) to consider that reloading the segment wouldn't help us?

Because if I understand the issue in the first place, the difference was `0.450+sec` no?

Reply:
This isn't the difference between its current buffered start of the segment and what we expected it to be (which is `MAX_TIME_MISSING_FROM_COMPLETE_SEGMENT`, also used in this same file elsewhere, to detect GCed segments), this is the difference between its current buffered start and - only if it was already pushed before - its previous initial buffered start.

The idea is:
  1. we find out that our segment is garbage collected (relying on `MAX_TIME_MISSING_FROM_COMPLETE_SEGMENT`, which is set at 150 milliseconds today)
  2. we check if that segment has already been pushed in recent history:
        - if it hasn't (it's the first time we're loading this segment in recent history), we re-download it
        - if it has, we compare the initial `bufferedStart` value the last two times it has been pushed.
           If it appears to be very close both times. we infer that it may either be a browser or content issue and do not try to download anymore

I thought that it may be too specific to put it in the config (and I have no idea how I would name that variable :p)

Still I just pushed a commit with a comment that tries to better explain what those checks are.
---

Anyway, your comment helped me find an issue, in that we never checked if the current `bufferedStart` value (the currently buffered start time of the segment) was actually close to `entry.bufferedStart` - the __initial__ `bufferedStart` value after it has been pushed. We could thus have false negatives because we only checked initial values - which have a high chance to be equal in normal conditions.

This has been fixed in the same new commit. I also added comment to that one if that's not too clear.